kedro/framework/cli/starters.py

```diff
@@ -91,23 +162,20 @@ def new(
             "Cannot use the --directory flag without a --starter value."
         )
 
-    # The `astro-iris` was renamed to `astro-airflow-iris`, but old (external) documentation
-    # and tutorials still refer to `astro-iris`. The below line checks if a user has entered old
-    # `astro-iris` as the starter name and changes it to `astro-airflow-iris`.
-    starter_name = (
-        "astro-airflow-iris" if starter_name == "astro-iris" else starter_name
-    )
-    if starter_name in _STARTER_ALIASES:
+    starter_aliases = _get_starters_aliases()
+
+    if starter_name in starter_aliases:
         if directory:
             raise KedroCliError(
                 "Cannot use the --directory flag with a --starter alias."
             )
-        template_path = _STARTERS_REPO
-        directory = starter_name
-        checkout = checkout or version
+        template_path = starter_aliases[starter_name]["template_path"]
+        # "directory" is an optional key for starters from plugins, so if the key is not present we will use "None".
+        directory = starter_aliases[starter_name].get("directory")
+        checkout = checkout or starter_aliases[starter_name].get("checkout", version)
     elif starter_name is not None:
         template_path = starter_name
-        checkout = checkout or version
+        checkout = checkout or starter_aliases[starter_name].get("checkout", version)
```

Comment:
```suggestion
        checkout = checkout or version
```
Whatever we decide to do with `checkout`, this part of the conditional should stay the same because it applies when `starter_name` isn't in `starter_aliases`.